File: app/src/main/java/fr/igm/robotmissions/ui/missions/MissionAdapter.java
package fr.igm.robotmissions.ui.missions;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import fr.igm.robotmissions.R;
import io.swagger.client.model.DeplacementMission;

public class MissionAdapter extends RecyclerView.Adapter<MissionAdapter.ViewHolder> {

    private List<DeplacementMission> missionList = new ArrayList<>();
    private MissionsFragment fragment;

    public MissionAdapter(MissionsFragment fragment) {
        super();
        this.fragment = fragment;
    }

    public void setMissionList(List<DeplacementMission> missionList) {
        this.missionList = missionList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_mission, parent, false);
        return new MissionAdapter.ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        DeplacementMission mission = missionList.get(position);

        // set item text
        holder.nameView.setText(mission.getName());
        holder.ifcNameView.setText(mission.getIfc().getName());
        holder.ifcFloorView.setText(mission.getFloor());
        // set detail intent click
        Context context = holder.cardView.getContext();
        Intent intent = new Intent(context, MissionDetailsActivity.class);
        intent.putExtra(MissionDetailsActivity.EXTRA_MISSION, mission);

        holder.cardView.setOnClickListener((_v) -> {
            fragment.startActivityForResult(intent, MissionsFragment.REQUEST_CODE);
        });
    }

    @Override
    public int getItemCount() {
        return missionList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final TextView ifcFloorView;
        private final View cardView;
        private final TextView nameView;
        private final TextView ifcNameView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            cardView = itemView.findViewById(R.id.mission_card_view);
            nameView = cardView.findViewById(R.id.mission_name_text);
            ifcNameView = cardView.findViewById(R.id.mission_ifc_target_name_text);
            ifcFloorView = cardView.findViewById(R.id.mission_ifc_floor);
        }
    }


}

File: app/src/main/java/fr/uge/robotsmissions/ui/missions/MissionAdapter.kt
package fr.uge.robotsmissions.ui.missions

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import fr.uge.robotsmissions.R
import fr.uge.robotsmissions.objects.Mission
import kotlinx.android.synthetic.main.item_mission.view.*
import java.text.SimpleDateFormat


class MissionAdapter(private val missions: ArrayList<Mission>, val itemClickListener: View.OnClickListener) :
    RecyclerView.Adapter<MissionAdapter.ViewHolder>() {


    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        val cardView = itemView.mission_card_view!!
        val nameView = cardView.mission_name_text!!
        val ifcView = cardView.mission_ifc_target_name_text!!
        var launchedView = cardView.mission_last_launched_text!!


        val clickListener = itemView.setOnClickListener(this)

        override fun onClick(v: View?) {
            var position: Int = layoutPosition
            println(position)
        }
    }

    override fun getItemCount(): Int {
        return missions.count()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MissionAdapter.ViewHolder {
        val viewItem = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_mission, parent, false)
        return MissionAdapter.ViewHolder(viewItem)
    }

    override fun onBindViewHolder(holder: MissionAdapter.ViewHolder, position: Int) {
        val mission = missions[position]
        val simpleDateFormat: SimpleDateFormat = SimpleDateFormat("dd MMM YYYY")
        holder.cardView.setOnClickListener(itemClickListener)

        holder.cardView.tag = position
        holder.nameView.text = mission.name
        holder.ifcView.text = "IFC Target : ${mission.ifc_target_name}"
        holder.launchedView.text =
            "Last launched : ${simpleDateFormat.format(mission.last_launched)}"

    }

    fun removeAt(position: Int) {
        missions.removeAt(position)
        notifyItemRemoved(position)
    }

    fun isEnabled(position: Int): Boolean {
        return true
    }

}

